gui/wxpython/gui_core/menu.py

```diff
@@ -305,3 +308,56 @@ def OnItemSelected(self, node):
             label = data['description']
 
         self.showNotification.emit(message=label)
+
+
+class FileHistory:
+    """Add recent files history menu"""
+
+    file_history = '.file_history'
+
+    def __init__(
+            self, app_name, parent_menu, pos,
+            history_len=10, text_widget=None,
+    ):
+        self.text_widget = text_widget
+        self.filehistory = wx.FileHistory(maxFiles=history_len)
+        self.config = wx.FileConfig(
+            style=wx.CONFIG_USE_LOCAL_FILE,
+            localFilename=os.path.join(
+                utils.GetSettingsPath(), self.file_history,
+            ),
+        )
+        self.config.SetPath(strPath=app_name)
+        self.filehistory.Load(self.config)
+
+        self.recent = wx.Menu()
+        self.filehistory.UseMenu(self.recent)
+        self.filehistory.AddFilesToMenu()
+
+        parent_menu.Insert(
+            pos=pos, id=wx.ID_ANY, text=_('&Recent Files'),
+            submenu=self.recent,
+        )
+
+        self.recent.Bind(
+            wx.EVT_MENU_RANGE, self.OnFileHistory,
+            id=wx.ID_FILE1, id2=wx.ID_FILE + history_len,
+        )
+
+    def OnFileHistory(self, event):
+        file_num = event.GetId() - wx.ID_FILE1
+        self.path = self.filehistory.GetHistoryFile(file_num)
+        self.filehistory.AddFileToHistory(self.path)  # move up the list
```

Comment:
I was probably not clear enough, this is the part I think should be hidden. (Although it looks much better than before!)

Reply:
> > If no recent files, menu is empty with long empty list (wxGTK): ... But It looks properly rendered on wxMSW: ...
> 
> Not directly our problem then. Let's leave it as it is then.

I adjusted it so that the recent files menu is inserted if the number of items > 0.